src/main/java/bio/terra/service/dataset/DatasetTableDao.java

```diff
@@ -118,11 +122,22 @@ public void removeTables(UUID parentId, List<String> tableNames) {
     }
   }
 
-  private void createColumns(UUID tableId, Collection<Column> columns) {
+  private void createColumnsNewTable(UUID tableId, Collection<Column> columns) {
+    createColumns(tableId, columns, 0);
+  }
+
+  public void createColumnsExistingTable(UUID tableId, Collection<Column> columns) {
+    MapSqlParameterSource params = new MapSqlParameterSource();
+    params.addValue("table_id", tableId);
+    Integer maxOrdinal = jdbcTemplate.queryForObject(sqlGetMaxColumnOrdinal, params, Integer.class);
+    // Need to requireNonNull or else SpotBugs complains
```

Comment:
If you believe it will never be `null`, then assign it to `int`. You'll get an NPE if it is ever `null`, but spotbugs shouldn't complain.
```suggestion
    @SuppressWarnings("ConstantConditions")
    int maxOrdinal = jdbcTemplate.queryForObject(sqlGetMaxColumnOrdinal, params, Integer.class);
    // Need to requireNonNull or else SpotBugs complains
```

Reply:
The default value is 0 in the database so it should never be `null`: https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/resources/db/changesets/20220202_adddatasetandsnapshottablecolumnorder.yaml#L11

Reply:
Hmm I still get an error from Spotbugs after that change: `DatasetTableDao.java:132:132 NP: Possible null pointer dereference in bio.terra.service.dataset.DatasetTableDao.createColumnsExistingTable(UUID, Collection) due to return value of called method (M)`